internal/ethapi/api_extra.go

```diff
@@ -92,3 +92,40 @@ func (s *BlockChainAPI) GetBadBlocks(ctx context.Context) ([]*BadBlockArgs, erro
 	}
 	return results, nil
 }
+
+// isHistoricalStateQueryAllowed returns a nil error only if:
+//   - the node is configured to accept historical proofs queries; or
+//   - the block number given is within the window of recent accepted
+//     blocks and not yet part of the historical blocks.
+//
+// Otherwise, a non-nil error is returned.
+func (s *BlockChainAPI) isHistoricalStateQueryAllowed(blockNumOrHash rpc.BlockNumberOrHash) (err error) {
+	historicalProofs, recentBlocksWindow := s.b.HistoricalConfig()
+	if historicalProofs {
+		return nil
+	}
+
+	lastBlock := s.b.LastAcceptedBlock()
+	lastNumber := lastBlock.NumberU64()
+
+	var number uint64
+	if blockNumOrHash.BlockNumber != nil {
+		number = uint64(blockNumOrHash.BlockNumber.Int64())
+	} else {
+		block, err := s.b.BlockByNumberOrHash(context.Background(), blockNumOrHash)
+		if err != nil {
+			return fmt.Errorf("getting block from hash: %s", err)
```

Comment:
again this does not seem like an error but rather a process. can we just return the err or wrap it like "failed to get block: %w"?

also "%w" verb wraps the error [see](https://pkg.go.dev/fmt#Errorf)

Reply:
See our little discussion at https://github.com/ava-labs/libevm/pull/92#discussion_r1889940968 😉

Reply:
seems more consistent with the code base to keep it "failed to: ..." or "could not ..." even if it's technically redundant language since it's an error.